module_utils/network_lsr/argument_validator.py

```diff
@@ -1858,6 +1904,31 @@ def _validate_post(self, value, name, result):
     VALIDATE_ONE_MODE_NM = "nm"
     VALIDATE_ONE_MODE_INITSCRIPTS = "initscripts"
 
+    def validate_route_tables(self, connection, idx):
+        table_cache = {}
+        for r in connection["ip"]["route"]:
+            if isinstance(r["table"], Util.STRING_TYPE):
+                tableid_find = False
+                if table_cache != {}:
+                    if r["table"] in table_cache.keys():
+                        tableid_find = True
+                        r["table"] = int(table_cache[r["table"]])
+                else:
+                    with open("/etc/iproute2/rt_tables", "r") as f:
```

Comment:
the parsing of the mapping should go to a separate function (that can be unit-tested)... and to unit-test it, you might need two functions: one that operates only on the `bytes file_content` (so it can be tested), and one that reads the file and passes it to the other function.